h/stcbackground.h

```diff
@@ -0,0 +1,48 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which 
+  accompanies this distribution, and is available at 
+  https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#ifndef __STCBACKGROUND__
+#define __STCBACKGROUND__
+
+#include "stcbase.h"
+
+#define N_TASK_TABLE_ENTRIES 101
+#define STC_BACKGROUND_INTERVAL 10
+#define LEN_TASK_LABEL 30
+
+typedef void (*Task)(void* server, void* taskInput); 
```

Comment:
We need to design this prototype carefully because we won't be able to change it without breaking backward compatibility. 
So, perhaps:
```c
typedef int (STCIntervalCallback)(struct STCBase_tag *stcBase,
                                  struct STCModule_tag *module, 
                                  struct STCIntervalCallbackData_tag *callbackData, 
                                  void *userData);
```

We should be able to derive all we need from these arguments.

All access to the fields in `callbackData` could be done via public functions, so we can change the structure without breaking old plugins - they all access the fields using those functions which are exported from `zssServer` dynamically at runtime.

For example, in your concrete interval callback you could do something like this even if we shuffle the fields in `struct STCIntervalCallbackData_tag`:
```c
static int checkDatasets(struct STCBase_tag *stcBase,
                         struct STCModule_tag *module, 
                         struct STCIntervalCallbackData_tag *callbackData, 
                         void *userData) {

  . . .
  if (stcIsIntervalTimedOut(callbackData)) {
    // issue a warning
  }

}
```

Reply:
modified

Reply:
just added one more extra argument in signature, `void* server`, as server may have context too for callbacks

```
typedef int (*STCIntervalCallback)(void* server,
                                  struct STCBase_tag *stcBase,
                                  struct STCModule_tag *module, 
                                  STCIntervalCallbackData* callbackData, 
                                  void *userData);
```

Reply:
moved struct definition to .c